From 2a0a94d753dfc674f49106a6b4629a17c834907a Mon Sep 17 00:00:00 2001
From: Michael Scott <mike@foundries.io>
Date: Fri, 31 Jul 2020 00:29:10 -0700
Subject: [PATCH 021/322] [FIO internal] common: FIOVB: split SPL config out

This allows removal of this driver when SPL is enabled.

Signed-off-by: Michael Scott <mike@foundries.io>
---
 common/Kconfig  | 8 ++++++++
 common/Makefile | 2 +-
 2 files changed, 9 insertions(+), 1 deletion(-)

diff --git a/common/Kconfig b/common/Kconfig
index f43a807654f..c5df066a695 100644
--- a/common/Kconfig
+++ b/common/Kconfig
@@ -596,6 +596,14 @@ config FIOVB
 	 This option enables compilation of Foundries.IO verified boot
 	 operations
 
+config SPL_FIOVB
+	bool "Build Foundries.IO Verified Boot operations"
+	depends on PARTITION_UUIDS
+	depends on OPTEE
+	help
+	 This option enables compilation of Foundries.IO verified boot
+	 operations
+
 config SPL_HASH
 	bool # "Support hashing API (SHA1, SHA256, etc.)"
 	help
diff --git a/common/Makefile b/common/Makefile
index 713fb0064de..e7d0d79aa72 100644
--- a/common/Makefile
+++ b/common/Makefile
@@ -137,4 +137,4 @@ obj-$(CONFIG_CMD_LOADB) += xyzModem.o
 obj-$(CONFIG_$(SPL_TPL_)YMODEM_SUPPORT) += xyzModem.o
 
 obj-$(CONFIG_AVB_VERIFY) += avb_verify.o
-obj-$(CONFIG_FIOVB) += fiovb.o
+obj-$(CONFIG_$(SPL_)FIOVB) += fiovb.o
-- 
2.34.1

